sensor/kubernetes/listener/resources/networkpolicy_test.go

```diff
@@ -0,0 +1,445 @@
+				Name:      "deploy-3",
+				Id:        "3",
+				Namespace: "secure",
+				PodLabels: map[string]string{
+					"app":  "sensor",
+					"role": "backend",
+				},
+			},
+		},
+		{
+			Deployment: &storage.Deployment{
+				Name:      "deploy-4",
+				Id:        "4",
+				Namespace: "default",
+				PodLabels: map[string]string{
+					"app": "sensor-2",
+				},
+			},
+		},
+	}
+
+	for _, d := range deployments {
+		ds.addOrUpdateDeployment(d)
+	}
+
+	cases := []struct {
+		name                string
+		netpol              *storage.NetworkPolicy
+		sel                 []map[string]string
+		expectedDeployments []*deploymentWrap
+	}{
+		{
+			name: "New NetworkPolicy",
+			netpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "default",
+			},
+			sel: []map[string]string{
+				{
+					"app":  "sensor",
+					"role": "backend",
+				},
+			},
+			expectedDeployments: []*deploymentWrap{
+				{
+					Deployment: &storage.Deployment{
+						Id:        "1",
+						Namespace: "default",
+					},
+				},
+			},
+		},
+		{
+			name: "Empty selector",
+			netpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "default",
+			},
+			sel: []map[string]string{},
+			expectedDeployments: []*deploymentWrap{
+				{
+					Deployment: &storage.Deployment{
+						Id:        "1",
+						Namespace: "default",
+					},
+				},
+				{
+					Deployment: &storage.Deployment{
+						Id:        "2",
+						Namespace: "default",
+					},
+				},
+				{
+					Deployment: &storage.Deployment{
+						Id:        "4",
+						Namespace: "default",
+					},
+				},
+			},
+		},
+		{
+			name: "Selector with no deployments",
+			netpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "default",
+			},
+			sel: []map[string]string{
+				{
+					"app": "central",
+				},
+			},
+			expectedDeployments: []*deploymentWrap{},
+		},
+		{
+			name: "Namespace with no deployments",
+			netpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "random_namespace",
+			},
+			sel: []map[string]string{
+				{
+					"app": "sensor",
+				},
+			},
+			expectedDeployments: []*deploymentWrap{},
+		},
+		{
+			name: "Namespace with no deployments, no selector",
+			netpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "random_namespace",
+			},
+			sel:                 []map[string]string{},
+			expectedDeployments: []*deploymentWrap{},
+		},
+		{
+			name: "Disjunction selector",
+			netpol: &storage.NetworkPolicy{
+				Id:        "1",
+				Namespace: "default",
+			},
+			sel: []map[string]string{
+				{
+					"app":  "sensor",
+					"role": "backend",
+				},
+				{
+					"app": "sensor-2",
+				},
+			},
+			expectedDeployments: []*deploymentWrap{
+				{
+					Deployment: &storage.Deployment{
+						Id:        "1",
+						Namespace: "default",
+					},
+				},
+				{
+					Deployment: &storage.Deployment{
+						Id:        "4",
+						Namespace: "default",
+					},
+				},
+			},
+		},
+	}
+	for _, c := range cases {
```

Comment:
I'd suggest you wrap all the content inside the `for` loop in a subtest. Otherwise it's difficult to figure out which test failed if there is a failure.
```go
for _, c := range cases {
		t.Run(c.name, func(t *testing.T) {
			// ... inner loop code here
		})
}
```

Plus, that's a very useful usage of the `name` property you already added to the test case struct.

Reply:
This is valid for the other test as well.

Reply:
Yeah, you're right. I've implemented this in both tests.